### PR TITLE
Add is_modified method to Document

### PR DESCRIPTION
**Pull request by lafrech**
I would like to be able to tell if a Document was modified.

Was there a simpler / more recommended way?

I suppose we could also use 

    if self._data._to_mongo_update() is not None

but it takes more computations.

I could add tests if you're interested.

### CODE REVIEW

**Comment by touilleMan** [comment]
This is a pretty good idea ! Beside it adds coherence with `EmbeddedDocument`.

Can you provide tests for this in `test_document.py` ?

**Comment by lafrech** [comment]
Great. I'll try to do that.

**Comment by lafrech** [comment]
I just added a few tests.

File: tests/test_document.py [hunk 1]
```diff
@@ -230,6 +230,36 @@ class CustomIdInheritance(CustomId):
...
+        @self.instance.register
+        class Vehicle(EmbeddedDocument):
+            name = fields.StrField()
+
+        @self.instance.register
+        class Driver(Document):
+            name = fields.StrField()
+            vehicle = fields.EmbeddedField(Vehicle)
+
+        driver = Driver(name='Marty')
```
Comment (touilleMan) [inline tests/test_document.py]: This document has not been commited yet, thus it should be `is_modified() == True`

You should check `is_created` within `is_modified` for this.

**Comment by touilleMan** [comment]
Added a comment about the need to use `is_created` inside `is_modified`

**Comment by lafrech** [comment]
> Added a comment about the need to use is_created inside is_modified

Yep. I get notified of those comments as well.

> This document has not been commited yet, thus it should be is_modified() == True
> 
> You should check is_created within is_modified for this.

Right. I pushed a fix.

**Comment by touilleMan** [comment]
Good job ! :+1:

**Pull request merged**